obspy/io/cmtsolution/core.py

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+CMTSOLUTION file format support for ObsPy.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import math
+import uuid
+import warnings
+
+
+from obspy import UTCDateTime
+from obspy.core.event import (Catalog, Event, EventDescription, Origin,
+                              Magnitude, FocalMechanism, MomentTensor, Tensor,
+                              SourceTimeFunction)
+from obspy.geodetics import FlinnEngdahl
+
+fe = FlinnEngdahl()
+
+
+def _get_resource_id(cmtname, res_type, tag=None):
+    """
+    Helper function to create consistent resource ids.
+    """
+    res_id = "smi:local/cmtsolution/%s/%s" % (cmtname, res_type)
+    if tag is not None:
+        res_id += "#" + tag
+    return res_id
+
+
+def _buffer_proxy(filename_or_buf, function, reset_fp=True,
```

Comment:
Yea. Eventually we really need to execute #952...but this here seems like a fairly nice way to do it.